pkg/traces/config.go

```diff
@@ -157,11 +157,16 @@ func (r *ReceiverMap) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	for k := range *r {
 		if strings.HasPrefix(k, otlpReceiverName) {
 			// for http and grpc receivers, include_metadata is set to true by default
-			protocolsCfg, ok := (*r)[k].(map[interface{}]interface{})["protocols"].(map[interface{}]interface{})
+			receiverCfg, ok := (*r)[k].(map[interface{}]interface{})
 			if !ok {
 				return fmt.Errorf("failed to parse OTLP receiver config: %s", k)
 			}
 
+			protocolsCfg, ok := receiverCfg["protocols"].(map[interface{}]interface{})
+			if !ok {
+				return fmt.Errorf("failed to parse OTLP receiver protocol config: %s", k)
```

Comment:
Can this error message explain why the config is invalid so users know what to fix?

Reply:
Good idea, but I wasn't sure what would be an accurate error message. We could say something like "no protocol config found", but I'm not sure if this would be accurate all the time. For example, if the type assertion failed because the yaml has a string instead of a map - then the protocol info isn't empty, but it's not what we expected (a map). Maybe an error like "no OTLP receiver protocol information in a valid key:value format was found"? Or maybe somehow we can report the line in the yaml config which has a problem?

Reply:
A generic message mentioning the requirements is probably sufficient: 

> otlp receiver requires a "protocol" field which must be a YAML map

Reply:
Thank you, I changed it to the error message you said and I do like it a lot more now. Just one thing - I wrote it as "protocols" instead of "protocol", I think it's plural in the config.